src/tqec/circuit/qubit_map.py

```diff
@@ -116,6 +116,30 @@ def to_circuit(self) -> stim.Circuit:
     def __getitem__(self, index: GridQubit) -> int:
         return self.q2i[index]
 
+    def to_dict(self) -> dict[str, Any]:
+        """Return a dictionary representation of the qubit map.
+
+        The dictionary is intended to be used as a JSON object.
+        """
+        return {
+            "i2q": [[qi, q.to_dict()] for qi, q in self.i2q.items()],
+            "q2i": [[q.to_dict(), qi] for qi, q in self.i2q.items()],
```

Comment:
This is duplicated information. You can store only `i2q` and that should be enough.